resources/clients/react/elros-pam/src/FileUpload.tsx

```diff
@@ -6,63 +6,66 @@ import {
   SpaceBetween,
   Spinner,
 } from "@cloudscape-design/components";
-import { ChangeEvent, useRef, useState } from "react";
+import { ChangeEvent, useCallback, useRef, useState } from "react";
 
 export interface FileUploadProps {
   accept?: string[];
   disabled: boolean | undefined;
   onSubmit: (file: File) => Promise<void>;
 }
 
-function FileUpload({ disabled, onSubmit, accept }: FileUploadProps) {
+function FileUpload({ disabled, onSubmit, accept = [] }: FileUploadProps) {
   const [modalVisible, setModalVisible] = useState(false);
   const [isLoading, setIsLoading] = useState(false);
   const [selectedFile, setSelectedFile] = useState<File | null>(null);
   const [error, setError] = useState<string | null>(null);
   const fileInput = useRef<HTMLInputElement>(null);
 
-  const handleChange = ({ target }: ChangeEvent<HTMLInputElement>) => {
-    if (target.files?.length) {
-      setSelectedFile(target.files[0]);
-    }
-  };
+  const handleChange = useCallback(
```

Comment:
I haven't gotten a grip on when useCallback should be employed. Can you explain?

Reply:
useCallback ensures a single reference to the function, so my general guidance is arrow functions for internal callbacks, useCallback for passing it in props so the child doesn't rerender.